2024/day2/part2.js

```diff
@@ -0,0 +1,43 @@
+import utils from '../../utils/index.js';
+
+const year = 2024;
+const day = 2;
+const part = 'B';
+let sum = 0;
+
+const input = utils.getInput();
+
+for (let line of input) {
+    const levelArray = line.split(" ");
+    let validLevel = true;
+    validLevel = validateLevel(levelArray);
+    sum += validLevel ? 1 : 0;
+}
+
+function validateLevel(levelArray, backupFlag) {
+    let validLevel = true;
+    let initialDirection;
+    for (let i = 0; i < levelArray.length - 1 && validLevel == true; i++) {
+        let currentDirection;
+        if (i == 0) {
+            initialDirection = (+levelArray[i] < +levelArray[i + 1]) ? 'up' : 'down';
+        } else {
+            currentDirection = (+levelArray[i] < +levelArray[i + 1]) ? 'up' : 'down';
+        }
```

Comment:
This logic for calculating a direction is done multiple times. Probably deserves its own function.